tests/type-tests.dx

```diff
@@ -100,12 +100,15 @@ xr = map IToF arr
 
 :t [1, [2]]
 > Type error:Couldn't synthesize a class dictionary for: (FromInteger ((Fin 1) => Int32))
+>
+> :t [1, [2]]
+>     ^
 
 :t [[1, 2], [3, 4]]
 > ((Fin 2) => (Fin 2) => Int32)
 
 :t [[1, 2], [3]]
-> Type error:Table literal has 1 elements, but its inferred index set (((Fin 2) => Int32)) has a size of 2
+> Type error:Table type doesn't match annotation
```

Comment:
Is there any reason why we can't give an equally good error message now? The new message is super unhelpful